### PR TITLE
New J2K: Convert Number.[type]Value() to Number.to[type]()

### PR DESCRIPTION
**Pull request by harry1453**
https://youtrack.jetbrains.com/issue/KT-35593

### CODE REVIEW

File: nj2k/testData/newJ2k/methodCallExpression/specialBuiltinMembers.kt [hunk 1]
```diff
@@ -71,4 +71,14 @@ internal class A {
...
     }
+
+    fun kt35593() {
+        val number: Number = 1
+        val b: Byte = number.byteValue()
+        val d: Double = number.doubleValue()
+        val f: Float = number.floatValue()
+        val i: Int = number.intValue()
+        val l: Long = number.longValue()
+        val s: Short = number.shortValue()
```
Comment (darthorimar) [inline nj2k/testData/newJ2k/methodCallExpression/specialBuiltinMembers.kt]: correct one is: 
```
   val b = number.toByte()
   val d = number.toDouble()
   val f = number.toFloat()
   val i = number.toInt()
   val l = number.toLong()
   val s = number.toShort()
```

**Comment by harry1453** [comment]
Oops, sorry! I've pushed up the correct test data now

**Comment by harry1453** [comment]
Sorry! I've merged it to be mostly up to date and fixed the testdata

**Comment by darthorimar** [comment]
Thank you!

**Pull request merged**